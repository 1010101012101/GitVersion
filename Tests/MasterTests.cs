using GitFlowVersion;
using NUnit.Framework;

[TestFixture]
public class MasterTests
{

    [Test]
    public void Commit_on_tag_should_return_tag_as_version()
    {
        var version = FinderWrapper.FindVersionForCommit("a682956dccae752aa24597a0f5cd939f93614509", "master");
        Assert.AreEqual(0, version.Major);
        Assert.AreEqual(1, version.Minor);
        Assert.AreEqual(0, version.Patch);
        Assert.AreEqual(Stage.Final, version.Stage);
        Assert.AreEqual(0, version.PreRelease);
    }

    [Test]
    public void Commit_in_front_of_tag_should_return_tag_as_version()
    {
        var version = FinderWrapper.FindVersionForCommit("6b503e747408bbbcac7ec20a6c81cf10e53b6dcd", "master");
        Assert.AreEqual(0, version.Major);
        Assert.AreEqual(1, version.Minor);
        Assert.AreEqual(0, version.Patch);
        Assert.AreEqual(Stage.Final, version.Stage);
        Assert.AreEqual(0, version.PreRelease);
    }
}