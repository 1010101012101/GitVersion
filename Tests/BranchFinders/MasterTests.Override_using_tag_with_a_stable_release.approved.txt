{
    "Version": {
        "Suffix": null,
        "Major": 0,
        "Minor": 2,
        "Patch": 0,
        "PreReleaseTag": {
            "Name": null,
            "Number": null
        },
        "PreReleasePartTwo": null
    },
    "BranchType": 4,
    "BranchName": "master",
    "Sha": "000000000000000000000000000000000000000"
}